docs/diga_api.md

```diff
@@ -0,0 +1,289 @@
+
+> Die Vergabe, die Änderung der Daten [...] erfolgen auf Antrag. Der Erfassungsbeleg kann bei der ARGE•IK sowie bei jedem Sozialversicherungsträger angefordert oder auf der Internetseite der ARGE•IK (www.arge-ik.de) heruntergeladen werden.
+
+You need to request the IK or changes using a form.
+You can check the [ARGE-IK](https://www.dguv.de/arge-ik/index.jsp) website for more information (German).
+Specific information for requesting the number can be found in the [Antrag](https://www.dguv.de/arge-ik/antrag/index.jsp) tab.
+The form can be sent by email and a blueprint can be downloaded [here](https://www.dguv.de/medien/arge-ik/downloads/erfass.pdf).
+
+Additional information at [gkv page](https://www.gkv-datenaustausch.de/leistungserbringer/sonstige_leistungserbringer/sonstige_leistungserbringer.jsp) can be found in [20210401_Broschuere_TP5 (PDF)](https://www.gkv-datenaustausch.de/media/dokumente/leistungserbringer_1/sonstige_leistungserbringer/20210401_Broschuere_TP5.pdf)
+which also contains a section about requesting certificates from ITSG trust center.
+
+## Verfahren
+
+The `verfahren` parameter consists of 5 characters describing the kind of the request.
+The first character is either `T` or `E` denoting `Testdaten` (test data) or `Echtdaten` (real data) - [attachment 4 - Verfahrenskennungen](https://www.gkv-datenaustausch.de/media/dokumente/standards_und_normen/technische_spezifikationen/Anlage_4_-_Verfahrenskennungen.pdf) on [gkv technische standards page](https://www.gkv-datenaustausch.de/technische_standards_1/technische_standards.jsp).
+
+The next 3 characters are used to discriminate between:
+
+- verification - DFC (**D**iga **F**reischalt**c**ode)
+- billing - DRE (**D**iga **Re**chnung).
+
+The last one is for versioning which leaves us with the following values:
+
+- `EDFC0` and `TDFC0` for verification
+- `EDRE0` and `TDRE0` for billing.
+
+## Nutzdaten
+
+`nutzdaten` is the tricky field which contains verification or billing related information based on the `verfahren` value.
+`nutzdaten` are defined in an xml sheet which needs to be encrypted and encoded as byte array before sending the actual requests.
+The structure of the xml files are defined by `xsd` files which can be found in `src/main/resources` or downloaded from [gkv page](https://www.gkv-datenaustausch.de/leistungserbringer/digitale_gesundheitsanwendungen/digitale_gesundheitsanwendungen.jsp) as [attachment - DiGA-Abrechnungsschema (XSD) (ZIP)](https://www.gkv-datenaustausch.de/media/dokumente/leistungserbringer_1/digitale_gesundheitsanwendungen/technische_anlagen_aktuell_7/DiGA_XSD_20201029.zip).
+Examples for the data and additional details on the encryption are described in the following.
+
+# Request details
+
+## Code verification requests
+
+See `EDFC0-Pruefung_2.0.0.xsd` file for reference:
+
+Data for sending the request:
+
+```xml
```

Comment:
I am not 100% about this TBH. We dont have the client up and running yet and we dont have the certificates for encryption/decryption. Therefore, it's hard to give real examples. Would be great if this could be verified or correct. Same for the response data.